content/reader.go

```diff
@@ -142,3 +175,37 @@ func ensureEOF(r io.Reader) error {
 	}
 	return nil
 }
+
+// Decode a Hash object
```

Comment:
nit: same as above about starting with function name.

mainly matters if the plan is to submit as a PR upstream.